### ISSUE TITLE
Exception after update

### ISSUE THREAD

**Issue by Gouvernathor** [renpy/renpy#4829]
```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "renpy/common/000statements.rpy", line 670, in execute_call_screen
    store._return = renpy.call_screen(name, *args, **kwargs)
TypeError: 'float' object cannot be interpreted as an integer

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "launcher/game/front_page.rpyc", line 252, in script
  File "E:\dev\renpy-8-nightly-sdk\renpy\ast.py", line 2281, in execute
    self.call("execute")
  File "E:\dev\renpy-8-nightly-sdk\renpy\ast.py", line 2263, in call
    return renpy.statements.call(method, parsed, *args, **kwargs)
  File "E:\dev\renpy-8-nightly-sdk\renpy\statements.py", line 342, in call
    return method(parsed, *args, **kwargs)
  File "renpy/common/000statements.rpy", line 670, in execute_call_screen
    store._return = renpy.call_screen(name, *args, **kwargs)
  File "E:\dev\renpy-8-nightly-sdk\renpy\exports.py", line 3355, in call_screen
    rv = renpy.ui.interact(mouse="screen", type="screen", roll_forward=roll_forward)
  File "E:\dev\renpy-8-nightly-sdk\renpy\ui.py", line 299, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\core.py", line 2989, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, pause=pause, pause_start=pause_start, pause_modal=pause_modal, **kwargs) # type: ignore
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\core.py", line 3582, in interact_core
    self.draw_screen(root_widget, fullscreen_video, (not fullscreen_video) or video_frame_drawn)
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\core.py", line 2187, in draw_screen
    surftree = renpy.display.render.render_screen(
  File "render.pyx", line 492, in renpy.display.render.render_screen
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 886, in render
    surf = render(child, width, height, cst, cat)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 886, in render
    surf = render(child, width, height, cst, cat)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 886, in render
    surf = render(child, width, height, cst, cat)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\screen.py", line 731, in render
    child = renpy.display.render.render(self.child, w, h, st, at)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 886, in render
    surf = render(child, width, height, cst, cat)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 1347, in render
    surf = render(child,
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 1062, in render
    surf = render(d, rw, height - y, cst, cat)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 1347, in render
    surf = render(child,
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 1861, in render
    cwidth, top = sizeit('t', cwidth, height, cwidth, top)
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 1857, in sizeit
    rend = renpy.display.render.render_for_size(pos_d[pos], width, height, st, at)
  File "render.pyx", line 318, in renpy.display.render.render_for_size
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\layout.py", line 1382, in render
    back = render(style.background, bw, bh, st, at)
  File "render.pyx", line 170, in renpy.display.render.render
  File "render.pyx", line 266, in renpy.display.render.render
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\imagelike.py", line 422, in render
    self.draw_pattern(draw, left, top, right, bottom)
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\imagelike.py", line 442, in draw_pattern
    draw(left, -right, top, -bottom)
  File "E:\dev\renpy-8-nightly-sdk\renpy\display\imagelike.py", line 400, in draw
    for x in range(0, xtiles):
TypeError: 'float' object cannot be interpreted as an integer

Windows-10-10.0.19045 AMD64
Ren'Py 8.2.0.23070901+nightly
Ren'Py Launcher 8.2.0.23070901+nightly
Mon Jul 10 05:28:05 2023
```

**Comment by Gouvernathor** [comment]
The issue is the int casting that went away in a33009b3828b813bdbd75c2f4fd20a1cfb1676ca
I'm not sure how the issue should be solved considering the changes made @renpytom

**Comment by Gouvernathor** [comment]
fixed by be36f99
